assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -37,86 +48,81 @@ def _convert_uri_folder_to_mltable(
-        monitor_current_time: The current time of the window (inclusive).
-        window_size_in_days: Number of days from current time to start time window (exclusive).
-        preprocessed_data: The mltable path pointing to location where the outputted mltable will be written to.
-        extract_correlation_id: The boolean to extract correlation Id from the MDC logs.
-    """
-    # Format the dates
-    format_data = "%Y-%m-%d %H:%M:%S"
-    start_datetime = parse_datetime_from_string(format_data, data_window_start)
-    end_datetime = parse_datetime_from_string(format_data, data_window_end)
-
-    # Create mltable definition - extract, filter and convert columns.
-    table = _convert_uri_folder_to_mltable(start_datetime, end_datetime, input_data)
-
-    # Create MLTable in different location
-    save_path = tempfile.mktemp()
-    table.save(save_path)
-
-    # Save preprocessed_data MLTable to temp location
-    des_path = preprocessed_input_data + "temp"
-    fs = AzureMachineLearningFileSystem(des_path)
-    print("MLTable path:", des_path)
-    # TODO: Evaluate if we need to overwrite
-    fs.upload(
-        lpath=save_path,
-        rpath="",
-        **{"overwrite": "MERGE_WITH_OVERWRITE"},
-        recursive=True,
-    )
+def _convert_mltable_to_spark_df(table: MLTable, preprocessed_input_data: str, fs: AbstractFileSystem) -> DataFrame:
+    """Convert MLTable to Spark DataFrame."""
+    with tempfile.TemporaryDirectory() as mltable_temp_path:
+        # Save MLTable to temp location
+        table.save(mltable_temp_path)
+
+        # Save preprocessed_data MLTable to temp location
+        des_path = preprocessed_input_data + "temp"
+        fs = fs or AzureMachineLearningFileSystem(des_path)  # for testing
+        print("MLTable path:", des_path)
+        # TODO: Evaluate if we need to overwrite
+        # Richard: why we need to upload the mltable folder to azureml://?
+        fs.upload(
+            lpath=mltable_temp_path,
+            rpath=des_path,
+            **{"overwrite": "MERGE_WITH_OVERWRITE"},
+            recursive=True,
+        )
 
     # Read mltable from preprocessed_data
-    df = try_read_mltable_in_spark(des_path, "preprocessed_data")
+    return try_read_mltable_in_spark(des_path, "preprocessed_data")
 
-    if not df:
-        print("Skipping the Model Data Collector preprocessor.")
-        post_warning_event(
-            "Although data was found, the window for this current run contains no data. "
-            + "Please visit aka.ms/mlmonitoringhelp for more information."
-        )
-        return
+
+def _get_data_columns(df: DataFrame) -> list:
+    columns = []
+    if MDC_DATA_COLUMN in df.columns:
+        columns.append(MDC_DATA_COLUMN)
+    if MDC_DATAREF_COLUMN in df.columns:
+        columns.append(MDC_DATAREF_COLUMN)
+
+    return columns
+
+
+def _extract_data_and_correlation_id(df: DataFrame, extract_correlation_id: bool, datastore: str) -> DataFrame:
```

Comment:
Please add some comments to this fuction, in data case, the data is returned, while for dataref, the data path is returned